### PR TITLE
Print variable with different precision depending of LEGACY_UNITS

### PR DESCRIPTION
**Pull request by alkino**
Use more format instead of stringstream in codegen.

### CODE REVIEW

File: src/codegen/codegen_c_visitor.cpp [hunk 1]
```diff
@@ -986,17 +986,17 @@ std::vector<SymbolType> CodegenCVisitor::get_shadow_variables() {
 /****************************************************************************************/
 
 std::string CodegenCVisitor::get_parameter_str(const ParamVector& params) {
-    std::stringstream param_ss;
+    std::string param{};
```
Comment (iomaganaris) [inline src/codegen/codegen_c_visitor.cpp:989]: I am curious. Why does this function needs to change to using string as well?

Comment (alkino) [reply src/codegen/codegen_c_visitor.cpp:989]: it does not have to. But this is the last one using stringstream in this file, all other methods use fmt.

Comment (iomaganaris) [reply src/codegen/codegen_c_visitor.cpp:989]: I see. Thanks for the answer!

**Review by iomaganaris — APPROVED** [review]
LGTM

File: src/codegen/codegen_c_visitor.cpp [hunk 2]
```diff
@@ -1994,10 +1994,17 @@ void CodegenCVisitor::print_nmodl_constants() {
...
-            ss << it->get_value()->get_value() << ";";
-            printer->add_line(ss.str());
+#ifndef USE_LEGACY_UNITS
+#define USE_LEGACY_UNITS 0
+#endif
+#if USE_LEGACY_UNITS
+            std::string format_string = "static const double {} = {:g};";
+#else
+            std::string format_string = "static const double {} = {:.18g};";
+#endif
```
Comment (pramodk) [inline src/codegen/codegen_c_visitor.cpp:2004]: Just a `nitpick` : you can directly use `#if defined(USE_LEGACY_UNITS)`  here?

Comment (alkino) [reply src/codegen/codegen_c_visitor.cpp:2004]: No because it can be defined to `0`

Comment (pramodk) [reply src/codegen/codegen_c_visitor.cpp:2004]: Ok. I see now in CMakeLists.txt it's either 0 to 1 i.e. always defined. Why we need:
```
#ifndef USE_LEGACY_UNITS
 #define USE_LEGACY_UNITS 0
 #endif
```
?

**Review by pramodk — APPROVED** [review]

**Comment by pramodk** [comment]
I also make travis optional because its taking too long!

**Pull request merged**